study-builder/tenants/register-user-in-pepper.js

```diff
@@ -136,6 +136,20 @@ function (user, context, callback) {
             console.log('No Registration Mode passed in request');
         }
 
+        if (pepper_params.mode && pepper_params.mode === "login" &&
+            !user.app_metadata.user_guid &&
+            !pepper_params.tempUserGuid
```

Comment:
Could you double-check that we want && connecting everything here? Is it possible that we need is:
```javascript
if (pepper_params.mode && pepper_params.mode === "login" && (!user.app_metadata.user_guid ||
            !pepper_params.tempUserGuid)
```

Reply:
During login if it doesn't contain both
user.app_metadata.user_guid or pepper_params.tempUserGuid, then we consider it as NOT registered. 

Your expressions will be true even if one of them doesn't exist !